app/src/main/java/ro/code4/monitorizarevot/ui/forms/FormsViewModel.kt

```diff
@@ -16,21 +16,21 @@ import ro.code4.monitorizarevot.data.model.Question
 import ro.code4.monitorizarevot.data.pojo.AnsweredQuestionPOJO
 import ro.code4.monitorizarevot.data.pojo.FormWithSections
 import ro.code4.monitorizarevot.data.pojo.PollingStationInfo
+import ro.code4.monitorizarevot.helper.*
 import ro.code4.monitorizarevot.helper.Constants.REMOTE_CONFIG_FILTER_DIASPORA_FORMS
-import ro.code4.monitorizarevot.helper.completedPollingStationConfig
-import ro.code4.monitorizarevot.helper.zipLiveData
 import ro.code4.monitorizarevot.ui.base.BaseFormViewModel
 
 class FormsViewModel : BaseFormViewModel() {
-    private val formsLiveData = MutableLiveData<ArrayList<ListItem>>()
+    private val formsLiveData = MutableLiveData<Result<ArrayList<ListItem>>>()
     private val selectedFormLiveData = MutableLiveData<FormDetails>()
     private val selectedQuestionLiveData = MutableLiveData<Pair<FormDetails, Question>>()
     private val syncVisibilityLiveData = MediatorLiveData<Int>()
     private val navigateToNotesLiveData = MutableLiveData<Question?>()
-    private val pollingStationLiveData = MutableLiveData<PollingStationInfo>()
+    private val pollingStationLiveData = MutableLiveData<Result<PollingStationInfo>>()
 
     init {
         getForms()
+        //todo this does not work as expected.
```

Comment:
Are you referring to not handling exceptions when fetching the polling station text?

Reply:
It is connected with the comments from below where there is a null value in an rxjava operator. When this method is called for the first time, and neither api nor the db return something, the null values do not behave as expected. In RxJava1 it was valid, however, in rxjavsa2 it is not.